lib/usd/ui/layerEditor/batchSaveLayersUIDelegate.cpp

```diff
@@ -38,15 +38,15 @@ MayaUsd::BatchSaveResult UsdLayerEditor::batchSaveLayersUIDelegate(const MDagPat
         if (MayaUsd::utils::kSaveToUSDFiles == opt) {
             UsdLayerEditor::SaveLayersDialog dlg(nullptr, proxyShapes);
 
-            if (QDialog::Rejected == dlg.exec()) {
-                return MayaUsd::kAbort;
-            }
-
-            if (!dlg.layersNotSaved().isEmpty() || !dlg.layersWithErrorPairs().isEmpty()) {
-                return MayaUsd::kPartiallyCompleted;
-            } else {
-                return MayaUsd::kCompleted;
-            }
+            // The SaveLayers dialog only handles choosing new names for anonymous layers and making
+            // sure that they are remapped correctly in either their parent layer or by the owning
+            // proxy shape. The SaveLayers dialog itself does not currently handle the saving of
+            // file-backed layers, so for now we will return that we only partiall completed saving.
+            // This will trigger the LayerManager to double check what needs to be saved and to
+            // complete the saving of all file-backed layers.
+            //
+            return (QDialog::Rejected == dlg.exec()) ? MayaUsd::kAbort
+                                                     : MayaUsd::kPartiallyCompleted;
```

Comment:
I was returning that we had successfully saved everything if there weren't any errors, but the Save Layers dialog doesn't actually handle the file-backed layers.  The Layer Editor does when saved through there, but the code is not currently in the Dialog with the code to handle anonymous layers.  This is something I think we could continue to refactor but for now I'll just return that we only partially handled it.  This will indicate to the Layer Manager that it should check what else needs to be handled.

My automated tests did not hit this situation because we only go through this code path when the UI is running and we don't currently have automated tests for that.